pkg/kritis/crd/securitypolicy/violation.go

```diff
@@ -22,48 +22,54 @@ import (
 	"github.com/grafeas/kritis/pkg/kritis/apis/kritis/v1beta1"
 	"github.com/grafeas/kritis/pkg/kritis/constants"
 	"github.com/grafeas/kritis/pkg/kritis/metadata"
-)
-
-type Reason string
-
-// A list of security policy violations
-// TODO: Add Attestation checking violations
-const (
-	UnqualifiedImageViolation int = iota
-	FixUnavailableViolation
-	SeverityViolation
+	"github.com/grafeas/kritis/pkg/kritis/policy"
 )
 
 // Violation represents a vulnerability that violates an ISP
 type Violation struct {
 	Vulnerability metadata.Vulnerability
-	Violation     int
-	Reason        Reason
+	VType         policy.ViolationType
+	Msg           policy.Reason
+}
+
+// Reason returns the reason
+func (v Violation) Reason() policy.Reason {
```

Comment:
Public getters for public struct fields seems unusual. Structs should either have public fields, or public getters, not both. 

In this particular case, I don't understand why we introduce the additional overhead of implementing getters for a perfectly usable public field, unless there are bigger plans here.

Reply:
Since this struct implements an interface i did this. 
I was debating if i should make  the fields private and then add a New function (since its used across package boundaries). i can do that.